### ISSUE TITLE
Detecting user-actionable camera issues (e.g., camera shutters)

### ISSUE THREAD

**Issue by guidou** [w3c/mediacapture-main#930]
Devices sometimes allow users to disable a camera via various means (e.g, physical camera shutters, ).
In many cases users want to use the camera but have forgotten that they have disabled it by using a shutter, for example.
This is often detectable at the system level via some APIs, but on the Web it is not visible except as black frames or perhaps as a muted track.
Since tracks can be muted for any reason, an application cannot reliably provide feedback to the user with an explanation and possible ways to fix the problem. Black frames may also be produced by other non-system-level reasons.

It would be good to address this use case on the Web platform. A possible way would be to add an optional field to the mute event with a reason for the mute. Another possibility is to change the muted property from bool to an enum, or add a new  property or event to avoid breaking compatibility.

**Comment by eladalon1983** [comment]
This might be relevant: https://github.com/w3c/mediacapture-region/issues/9#issuecomment-1022334881
TL;DR: Another use for a MuteReason/MuteCause/mute-as-enum.

**Comment by eladalon1983** [comment]
Your thoughts about multiple concurrent reasons, btw? (See first link.)

**Comment by youennf** [comment]
From the user's perspective, the main issue is probably if user starts capturing with a camera that is shuttered.
In that case, couldn't it be the UA that would provide the warning to the user?
That would seem more robust than expecting every website to handle that case.

If user is using the shutter in the middle of a capture, user will probably not forget to remove it when needed.
I also somehow wonder whether user would like websites to know they activated the shutter in that case.

**Comment by eladalon1983** [comment]
The difference between the UA and the app is in information.

The UA always needs to be careful to not overcommunicate to the user. Should the user be informed that a shutter is active? How prominent should this warning be? A sane warning will invariably be discreet enough to be missed by some users.

The app, on the other hand, knows if camera-interaction is key to the current workflow. It can afford to communicate the presence of the shutter loudly and intrusively, in a way that cannot be missd.

**Comment by youennf** [comment]
It really depends of the flow and API shape we expose, let's assume the following:
1. If camera is shuttered, track is muted.
2. UA exposes the `requestUnmute` method we talked about (it might be good to make progress there).

Let's say that a web page is capturing but the camera track is muted.
The user clicks on the unmute button in the web page which calls `requestUnmute`.

In that particular case, the UA could tell the user what to do to unmute the track (unshutter the camera or provide appropriate information for that particular user setup).

I am not sure it is always best to expose muted reasons given they are open ended and/or OS specific.
It might make it hard for web pages to handle all cases/all platforms properly.

**Comment by jan-ivar** [comment]
Part of the privacy appeal (for me) of physical camera shutters is that they're not known to apps, hence apps can't refuse to work until I remove them. Exposing a JS API to the app the shutter is designed to block, seems like it would undermine that. It therefore seems superior to me to let the user agent handle this, without exposing this to the app.

**Comment by jan-ivar** [comment]
> UA exposes the `requestUnmute` method we talked about (it might be good to make progress there).

Is this https://github.com/w3c/mediacapture-extensions/issues/39? I'd love to make progress there.

**Comment by eladalon1983** [comment]
> apps can't refuse to work until I remove them.

Do you have a concrete example of such an app? (Other than patently malicious apps, where you should not allow cam/mic access to begin with.)
